### PR TITLE
Ignore unknown finalized/safe block in Taiko FCU during archive sync

### PR DESCRIPTION
**Pull request by flcl42**
FCU can contain head behind finalized block while syncing a taiko network in archive mode

## Changes

- Ignore unknown finalized/safe block hashes

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [x] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [x] Yes
- [ ] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Merge.Plugin/Handlers/ForkchoiceUpdatedHandler.cs [hunk 1]
```diff
@@ -165,16 +170,15 @@ protected virtual bool IsNewHeadAlignedWithChain(Block newHeadBlock, ForkchoiceS
 
         if (!blockInfo.WasProcessed)
         {
-            if (_blockTree.IsOnMainChainBehindHead(newHeadBlock))
+            if (!IsNewHeadAlignedWithChain(newHeadBlock, forkchoiceState, out ResultWrapper<ForkchoiceUpdatedV1Result>? errorResult))
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Merge.Plugin/Handlers/ForkchoiceUpdatedHandler.cs:173]: `IsNewHeadAlignedWithChain` is cryptic naming, can we name it better? Maybe would be easier if we would flip the bool semantics

**Review by LukaszRozmej — APPROVED** [review]
Only naming/readability issues

**Review by marcindsobczak — APPROVED** [review]

File: src/Nethermind/Nethermind.Merge.Plugin/Handlers/ForkchoiceUpdatedHandler.cs [hunk 2]
```diff
@@ -387,7 +391,7 @@ private void StartNewBeaconHeaderSync(ForkchoiceStateV1 forkchoiceState, BlockHe
         return branchOfBlocks;
     }
 
-    private BlockHeader? ValidateBlockHash(Hash256 blockHash, out string? errorMessage, bool skipZeroHash = true)
+    protected virtual BlockHeader? ValidateCheckpointBlockHash(ref Hash256 blockHash, out string? errorMessage, bool skipZeroHash = true)
```
[the remark this answers is not in the document]
Comment (flcl42) [reply src/Nethermind/Nethermind.Merge.Plugin/Handlers/ForkchoiceUpdatedHandler.cs:394]: ValidateConfirmedBlockHash?

[the remark this answers is not in the document]
Comment (flcl42) [reply src/Nethermind/Nethermind.Merge.Plugin/Handlers/ForkchoiceUpdatedHandler.cs:394]: ValidateSafeOrFinalizedBlockHash

File: src/Nethermind/Nethermind.Merge.Plugin/Handlers/ForkchoiceUpdatedHandler.cs (hunk 1 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Merge.Plugin/Handlers/ForkchoiceUpdatedHandler.cs:173]: already sounds easier to me

**Review by benaadams — APPROVED** [review]

**Review by benaadams — APPROVED** [review]

**Pull request merged**